control-services/cli/parser.py

```diff
@@ -68,24 +68,29 @@ def completeParser(self, services):
         self.__parser_task = self.__subparsers.add_parser(
             'task',
             description =   'Executes a task in Semaphore',
-            help =  'Executes a task in Semaphore'
+            help = 'Executes a task in Semaphore'
         )
 
         self.__parser_task.add_argument(
-            'name',
-            help = 'The name of the task to execute',
+            '--project',
+            dest = 'projectName',
+            help = 'The name of the project where the task will execute (e.g. "Class 21012" or "Management")',
         )
 
         self.__parser_task.add_argument(
-            'classID',
-            help = 'A class name formatted class#####',
-        )
-
-        self.__parser_task.add_argument(
-            'size',
-            help = 'The size of the class',
-        )
-
+            '--template',
+            dest = 'templateAlias',
+            help = 'The alias of the task template to execute (e.g. "Create Range"). Requires specifying the \
+                    project with the --project option',
+        )
+
+        # TODO: Currently not supported because semaphore does not allow passing parameters in to task template on a per-run basis
```

Comment:
On further review of the use cases for `vater task`, I don't see this feature fulfilling it's purpose until we can configure the arguments for the task from command line. We can't do something like `vater task --project management --template "create class"` and use it to create a class because I suspect it'll either A, just re create a class with the current values that were used to create the last class or B, won't make the class we need. Am I misunderstanding how this blocks #167 and the need to able to test Semaphore?

Reply:
Yes I agree and was under the same impression that passing the arguments from the command line is what is necessary to make this feature truly useful (and unblock #167). When I said "this isn't supported", I was actually referring to a task template's "arguments" parameter (shown below), which is where we currently specify the "class" and "size" values.
![image](https://user-images.githubusercontent.com/24918678/148957985-a95313a8-3d16-4505-8850-37ff22cf85f7.png)
This particular part of running a task from a template cannot be overwritten (the "arguments"), but I hadn't thought of looking up the template's environment itself and providing a new one, so thank you for the suggestion. I'll experiment with that today and see if it's a workable solution.

Reply:
Update: this is addressed in my new changes, however it is important to note that arguments provided directly to the task template (as pictured above) will overwrite arguments provided at the command line, and as far as I can tell there's nothing we can do about this. As a result, we should not create the templates with a default set of arguments

Reply:
I haven't reviewed the code, but how about grabbing the `-e` arguments and prompting for them or working them the same way I suggested earlier with the `environment`-- grab and append? Off the cuff idea before I review.

The GUI default arguments are important ease of use for instructors that will use the GUI to enter run time arguments, e.g. when creating a class. #95 attempts to offer a superior solution to the GUI issue and I think that maintaining GUI usability is more important than the command line at this time. When we demo / sell the product, the GUI is what we pitch because most day to day users won't touch the CLI.#174 

This might be the point where forking Semaphore is necessary to appropriately implement the features we need.

Reply:
With regards to
> but how about grabbing the -e arguments and prompting for them [...]
This is a good idea and was what I originally planned to do, but unfortunately the Semaphore does not give you a way to overwrite those arguments provided in the template. I agree that forking Semaphore and implementing the functionality through the UI that we _actually_ want might be the best way forward here, short of having the Command Line use one template and the GUI use a different template (which would be gross but would work)

Reply:
If we really want to push forward with this without working #95, as disgusting as it is, one work around might be to:

- Use the command line `vater task <project> <template> <args>` 
- Query for that task
- Build a new task with same parameters name "template-task-#" or something unique, but leave out the arguments
- Append, build a new argument list
- Run the task
- Delete the temporary task template

Reply:
Agreed this PR is one on track for what we want as far as GUI goes. It's developing really well

Reply:
Resolving this conversation as there is no further action for this particular PR